_posts/2021-01-12-github-application-running-on-faasd-behind-inlets.md

```diff
@@ -0,0 +1,508 @@
+$ export LICENSE="eyJhbGciOiJFUzI..."
+
+# Notice that this command is the output of the "inletsctl create" command above
+$ inlets-pro client --url "wss://XX.XXX.XXX.XX:8123/connect" \
+        --token "$TOKEN" \
+        --license "$LICENSE" \
+        --upstream $UPSTREAM \
+        --ports $PORTS
+```
+## Test
+In order to test it we need to install this app to selected repositories. Create a repository called "test-issues-bot", then install this app for it.
+![repository-access](/images/2021-01-12-github-application-using-go-and-inlets-pro/repository-access.png)
+
+Then, create an issue for the repository. You will see the message 
+> "Hello, issue opened by: developer-guy"
+
+![test-issue-bot](/images/2021-01-12-github-application-using-go-and-inlets-pro/test-issue-bot.png)
+
+Finally , let's close the issue by typing command */close*
+![close-issue](/images/2021-01-12-github-application-using-go-and-inlets-pro/close-issue.png)
+
+## Cleanup
+
+```bash
+$ multipass delete faasd
+$ multipass purge
+$ inletsctl delete --provider ec2 --id "YOUR_INSTANCE_ID" --access-token $AWS_ACCESS_KEY --secret-key $AWS_SECRET_KEY --region eu-central-1
+```
+
+## Further More
+You can run this Bot on a Raspberry Pi, with faasd and TLS with inlets-pro and Caddy!
+
+## Running on faasd running on a Raspberry Pi with TLS!
+
+> We are assuming you already have your SD card flashed for your Raspberry Pi with the latest version of Raspbian
+
+## Enable container features on your Pi
+To enable container features in the kernel, edit the `/boot/cmdline.txt` and add the following to the end of the line, making sure not to add any new lines to the end.
+```
+cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory
+```
+Now reboot your Pi
+
+## Get faasd
+We need the faasd source code to get started.
+
+```bash
+# Install git
+sudo apt update && \
+sudo apt install git -y
+
+# Clone the faasd repo to your Pi
+git clone https://github.com/openfaas/faasd.git
+```
+
+We can install faasd the easy way with as shell script, or do it the hard manual way. We're going to do it the easy way for this tutorial.
+```bash
+cd faasd
+sudo ./hack/install.sh
+```
+This shoud get faasd and all of it's dependecies up and running on your Pi.
+> You should get the password for the OpenFaaS gateway in the output, take a note of it.
+
+## Install inletsctl to for an exit node to expose our Pi to the internet!
+
+We're going to install inletsctl that would let us download inlets-pro on our Pi and create an exit-node on a cloud provider of our choice with inlets-pro server running as a systemd service, all of this just with a single command!
+You can learn more about inletsctl and the inlets project at [inlets.dev](https://inlets.dev)
+
+```bash
+# Install directly to /usr/local/bin/
+curl -sLSf https://inletsctl.inlets.dev | sudo sh
+```
+
+Lets' create an exit-node for our Pi on Google Compute Engine, that default to the f1-micro instance, making it almost free for us to run.
+
+```bash
```

Comment:
No need to repeat what's in the docs, and to complicate things by showing GCE with inlets. I'd like this section removed so the content is more focused on what we said we would do.

Reply:
done